### ISSUE TITLE
ORKScaleAnswerFormat doesn't display (some) long strings correctly

### ISSUE THREAD

**Issue by dilpreetsingh** [ResearchKit/ResearchKit#1043]
Getting some weird layout behaviour with ORKScaleAnswerFormat when assigning long minimum & maximum value description strings.

Here is the code for creating the answer format:
```
ORKScaleAnswerFormat(maximumValue: 10, minimumValue: 0, defaultValue: 5, step: 1, vertical: false, maximumValueDescription: "Can do it because of the pain", minimumValueDescription: "Can do it because of the pain")
```

Result (with `LAYOUT_DEBUG` enabled in ORKScaleSlider.m):

![simulator screen shot - iphone 8 - 2017-12-07 at 16 52 29](https://user-images.githubusercontent.com/4726826/33700730-12319dea-db6f-11e7-9216-ffe7ca526fb7.png)

The bug occurs in both master & stable branches. To re-create it, simply use the ORKCatalog sample app, and replace `step3AnswerFormat` in `scaleQuestionTask` method (`TaskListRow.swift`) with:

```
let step3AnswerFormat = ORKAnswerFormat.scale(withMaximumValue: 10, minimumValue: 1, defaultValue: NSIntegerMax, step: 1, vertical: false, maximumValueDescription: "Can do it because of the pain", minimumValueDescription: "Can do it because of the pain")
```

Run the app, select the **Scale Question** cell and step through.

**Some weird behaviour**:

Number 1:
If I replace the string "Can do it because of the pain problem" with "A somewhat long string to bring about" (note: both strings are 37 characters, but 8 and 7 words respectively), it works perfectly well:

![simulator screen shot - iphone 8 - 2017-12-07 at 16 43 47](https://user-images.githubusercontent.com/4726826/33700511-d71ae2c6-db6d-11e7-8870-aa95deed2294.png)

Number 2:
In the ORKCatalog app, replace the first answer format in the Scale Question task with the long string:

```
let step1AnswerFormat = ORKAnswerFormat.scale(withMaximumValue: 10, minimumValue: 1, defaultValue: NSIntegerMax, step: 1, vertical: false, maximumValueDescription: "Can do it because of the pain", minimumValueDescription: "Can do it because of the pain")
```

Run the app, and you'll see that the string appears correctly! But if you select an answer and press next, and then go back to the first step, it goes back to being cut off.

**Comment by srinathtm-apple** [comment]
Hi @dilpreetsingh , I am currently unable to reproduce this issue on iOS 11.1 running on an iPhone 7 plus with your steps. Can you also provide some details on what device/simulator and iOS build you are using while hitting this issue?

**Comment by dilpreetsingh** [comment]
Tested on a physical device: iPhone X running iOS 11.1.2
And simulator (which is in the screenshots): iPhone 8 running iOS 11.1
The ORKCatalog sample project i'm using is Version (and Build) 1.3

**Comment by dilpreetsingh** [comment]
I just tried with the iPhone 8 Plus simulator (iOS 11.1) and I can't reproduce the issue. Seems like the plus devices are possibly immune?

**Comment by srinathtm-apple** [comment]
Interesting, I tried on a 5s and that works fine as well. Anyways, I was able to repro this on the 8 and X simulators, looking into it.

**Comment by akshay-yadav-apple** [comment]
Hey @dilpreetsingh, 
We have a fix for this issue in #1046. 
Please confirm if it works for your use case.

**Comment by dilpreetsingh** [comment]
@akshay-yadav-apple  Works great! Thanks!
